app/model/Cors.scala

```diff
@@ -0,0 +1,28 @@
+package model
+
+import config.LoginConfig
+import play.api.Play.current
+import play.api.mvc.{RequestHeader, Result, Results}
+
+object Cors extends Results {
+
+  private val defaultAllowHeaders = List("X-Requested-With","Origin","Accept","Content-Type")
+  lazy val pandaDomainOption = play.api.Play.configuration.getString("pandomain.domain")
+
+  def apply(result: Result, allowedMethods: Option[String] = None)(implicit request: RequestHeader): Result = {
+
+    val responseHeaders = (defaultAllowHeaders ++ request.headers.get("Access-Control-Request-Headers").toList) mkString ","
+
+    request.headers.get("Origin") match {
+      case None => result
+      case Some(requestOrigin) if LoginConfig.isValidUrl(pandaDomainOption, requestOrigin)  => {
+        val headers = allowedMethods.map("Access-Control-Allow-Methods" -> _).toList ++ List(
+          "Access-Control-Allow-Origin" -> requestOrigin,
```

Comment:
It'd be slightly safer to return the host as obtained from the `pandaDomainOption`, rather than blindly echoing back the request.

There's no compiler-enforced check that the `if LoginConfig.isValidUrl` statement:
* is correct
* exists

This means it'd be quite easy to accidentally break the security check on this endpoint and doing so would compromise the privacy of all our readers. Given how serious the consequences are, it's better to code this a little more "defensively". By using the config to supply the CORS origin header, even if the check is broken malicious domains will be denied access.